src/FluentHub.Uwp/Views/MainPage.xaml.cs

```diff
@@ -84,17 +85,75 @@ protected override void OnNavigatingFrom(NavigatingCancelEventArgs e)
         }
 
-        private void OnSearchBarLostFocus(object sender, RoutedEventArgs e)
+        //private void OpenSearchAccelerator(KeyboardAcceleratorInvokedEventArgs e)
+        //{
+        //    SearchGitHubAutoSuggestBox.Visibility = Visibility.Visible;
+        //    SearchGitHubAutoSuggestBox.Focus(FocusState.Keyboard);
+        //    SearchGitHubButton.Visibility = Visibility.Collapsed;
+        //}
+
+        private void OnSearchGitHubAutoSuggestBoxLostFocus(object sender, RoutedEventArgs e)
+        {
+            SearchGitHubAutoSuggestBox.Visibility = Visibility.Collapsed;
+            SearchGitHubButton.Visibility = Visibility.Visible;
+        }
+
+        private void OnSearchGitHubAutoSuggestBoxQuerySubmitted(AutoSuggestBox sender, AutoSuggestBoxQuerySubmittedEventArgs args)
+        {
+            var parameter = args.QueryText;
+            navService.Navigate<Searches.RepositoriesPage>(new Models.FrameNavigationArgs() { Parameters = new() { parameter } });
+        }
+
+        private void OnSearchGitHubAutoSuggestBoxSuggestionChosen(AutoSuggestBox sender, AutoSuggestBoxSuggestionChosenEventArgs args)
         {
-            SearchBar.Visibility = Visibility.Collapsed;
-            SearchBarButton.Visibility = Visibility.Visible;
+            sender.Text = sender.Text;
+        }
+
+        private void OnSearchGitHubAutoSuggestBoxTextChanged(AutoSuggestBox sender, AutoSuggestBoxTextChangedEventArgs args)
+        {
+            // Only get results when it was a user typing, 
+            // otherwise assume the value got filled in by TextMemberPath 
+            // or the handler for SuggestionChosen.
+            if (args.Reason == AutoSuggestionBoxTextChangeReason.UserInput)
+            {
+                var suitableItems = new List<string>();
+
+                bool isInRepositories = false;
+                bool isInOrganizations = false;
+                bool isInUsers = false;
+
+                Type currentPage = navService.CurrentPage;
+                var currentPageNamespace = currentPage.ToString();
+
+                if (sender.Text != "")
+                {
+                    suitableItems.Add(sender.Text + " in all of Github");
+                }
+
+                if (currentPageNamespace.Contains("Views.Organizations"))
+                {
+                    suitableItems.Add(sender.Text + " in this organization");
+                }
+                else if (currentPageNamespace.Contains("Views.Repositories"))
+                {
+                    suitableItems.Add(sender.Text + " in this repository");
+
+                    // If that repository is in organization, it should be also added 'is this organization'
+                }
+                else if (currentPageNamespace.Contains("Views.Users"))
+                {
+                    suitableItems.Add(sender.Text + " in this user");
+                }
+
+                sender.ItemsSource = suitableItems;
```

Comment:
```diff
+                if (sender.Text != "")
+                {
+                    suitableItems.Add(sender.Text + " in all of Github");
+                }

+                if (currentPageNamespace.Contains("Views.Organizations"))
+                {
+                    suitableItems.Add(sender.Text + " in this organization");
+                }

+                else if (currentPageNamespace.Contains("Views.Repositories"))
+                {
+                    suitableItems.Add(sender.Text + " in this repository");
+                }

+                else if (currentPageNamespace.Contains("Views.Users"))
+                {
+                    suitableItems.Add(sender.Text + " in this user");
+                }
```
Should be the Label PrimerControl if possible

Reply:
Yeah. I really think so. Have to create DataTemplate(i do also not know whether i can or not).

Reply:
> Yeah. I really think so. Have to create DataTemplate(i do also not know whether i can or not).

Yeah I did think it was a bit far-fetched. If we did this, search results inside that box would not be able to appear, only those buttons aforementioned. 😕

Reply:
ACTUALLY--- I have an idea.

Reply:
> ACTUALLY--- I have an idea.

Perhaps I could implement buttons (amount per required) that would simultaneously appear/disappear depending on the context:
e.g. I'm searching inside FluentHub/FluentHub I will have 3 buttons, one for org, other for repo search and one for all github
e.g. I'm searching inside onein528/FluentReg I will have 3 buttons, one for user, other for repo search and one for all github

Reply:
@onein528

Reply:
Where would you like to display that 'Button'? In DataTemplate?

Reply:
> Where would you like to display that 'Button'? In DataTemplate?

Next to the SearchBar

Reply:
OK. Let's discuss later.

Reply:
Mine, done:

![image](https://user-images.githubusercontent.com/62196528/193416925-e2b9d8b7-b517-44ca-a684-adee718457c3.png)